File: akzo/akzo/report/recycle_report/recycle_report.py
# Copyright (c) 2023, parijat and contributors
# For license information, please see license.txt

# import frappe



import frappe
import datetime
def execute(filters=None):
	columns = [
	{'fieldname':'date','label':'Date','fieldtype':'Date'},
	{'fieldname':'filling','label':'Filling','fieldtype':'Int'},
	{'fieldname':'batch_makeing','label':'Batch Making','fieldtype':'Int'},
	{'fieldname':'percentage','label':'%','fieldtype':'Data'}
	]
	data = frappe.db.get_all('Recycle Report', ['date','filling','batch_makeing','percentage'])
#	chart = {'data':{'labels':['d','o','g','s'],'datasets':[{'values':[3,6,4,7]}]},'type':'bar'}
	chart = {
			'data':{
			'labels':['d','o','g','s'],
			'datasets':[
            #In axis-mixed charts you have to list the bar type first
			{'name':'Number','values':[3,6,4,7],'chartType':'bar'},
			{'name':'Vowel','values':[0,1,0,0],'chartType':'line'}
			]
			},
		'type':'axis-mixed'
		}
	return columns, data, chart

File: akzo/akzo/report/recycle_report/recycle_report.js
// Copyright (c) 2023, parijat and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["recycle report"] = {
	"filters": [

	]
};

frappe.query_reports["recycle report"] = {
        "formatter": function(value, row, column, data, default_formatter) {
                value = default_formatter(value, row, column, data);
                if (column.id == "percentage" && value < 90) {
                        value = "<span style='background-color:red;color:white;font-weight:bold;'>" + value + "</span>";
//                      var $value = $value + $(value).css("background-color", "#75ff3a");
//              value = $value.wrap("<p></p>").parent().html();
                }
                 if (column.id == "percentage" && value > 90) {

                value = "<span style='background-color:green;color:white;font-weight:bold;'>" + value + "</span>";
                }
                return value;
        },
};


